### PR TITLE
feat(cli): extend search command functionality

### LINKED ISSUE #1561: zli: refactor existing commands into the 'search' subcommand
See https://github.com/project-zot/zot/pull/1497#issuecomment-1591624653

### PR DESCRIPTION
**Pull request by laurentiuNiculae**
**What type of PR is this?**

**Which issue does this PR fix**:
Close #1561

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/cli/searcher.go [hunk 1]
```diff
@@ -770,7 +796,27 @@ func (search globalSearcherREST) search(config searchConfig) (bool, error) {
...
+}
+
+type derivedImageListSearcherREST struct{}
+
+func (search derivedImageListSearcherREST) search(config searchConfig) (bool, error) {
+	if !canSearch(config.params, newSet("derivedImage")) {
+		return false, nil
+	}
+
+	return true, fmt.Errorf("can't search by derived images: %w", zotErrors.ErrExtensionNotEnabled)
```
Comment (andaaron) [inline pkg/cli/searcher.go:809]:
```suggestion
	return true, fmt.Errorf("can't search for derived images: %w", zotErrors.ErrExtensionNotEnabled)
```

File: pkg/cli/searcher.go [hunk 2]
```diff
@@ -770,7 +796,27 @@ func (search globalSearcherREST) search(config searchConfig) (bool, error) {
...
+}
+
+type baseImageListSearcherREST struct{}
+
+func (search baseImageListSearcherREST) search(config searchConfig) (bool, error) {
+	if !canSearch(config.params, newSet("baseImage")) {
+		return false, nil
+	}
+
+	return true, fmt.Errorf("can't search by base images: %w", zotErrors.ErrExtensionNotEnabled)
```
Comment (andaaron) [inline pkg/cli/searcher.go:819]:
```suggestion
	return true, fmt.Errorf("can't search for base images: %w", zotErrors.ErrExtensionNotEnabled)
```

File: golangcilint.yaml [hunk 3]
```diff
@@ -52,6 +52,7 @@ linters-settings:
   gomoddirectives:
     replace-allow-list:
       - github.com/google/go-containerregistry
+      - github.com/opencontainers/image-spec
```
[the remark this answers is not in the document]
Comment (laurentiuNiculae) [reply golangcilint.yaml:55]: I don't know why but the linter is failing. It's telling me he can't parse the go.mod file and that there is a replace problem for that image

[the remark this answers is not in the document]
Comment (andaaron) [reply golangcilint.yaml:55]: Do you have a modified go.mod locally? For the other task related to artifacts?

[the remark this answers is not in the document]
Comment (laurentiuNiculae) [reply golangcilint.yaml:55]: I have the same go.mod file as I have pushed in this PR

[the remark this answers is not in the document]
Comment (laurentiuNiculae) [reply golangcilint.yaml:55]:
```
/home/laur/dev/zot/hack/tools/bin/golangci-lint --config ./golangcilint.yaml run --enable-all --out-format=colored-line-number --build-tags containers_image_openpgp ./...
WARN [runner/nolint] Found unknown linters in //nolint directives: go-staticcheck
go.mod:486:1: replacement are not allowed: github.com/google/go-containerregistry (gomoddirectives)
replace github.com/google/go-containerregistry => github.com/google/go-containerregistry v0.14.1-0.20230409045903-ed5c185df419
```

**Review by andaaron — APPROVED** [review]

**Comment by andaaron** [comment]
Superseded by https://github.com/project-zot/zot/pull/1674

**Pull request closed without merging**